src/handlers/source.rs

```diff
@@ -2,9 +2,24 @@ use super::error::ResourceError;
 
 pub async fn create_source(
     user_identity: String,
-    new_source: crate::entity::user::CalendarEventSource,
+    new_source: crate::entity::user::EventSource,
     state: actix_web::web::Data<crate::app::State>,
-) -> Result<crate::entity::user::CalendarEventSource, super::error::ResourceError> {
+) -> Result<crate::entity::user::EventSource, super::error::ResourceError> {
+    if new_source.name.len() == 0 {
+        return Err(ResourceError::InvalidInput("Name is empty".to_string()));
+    }
+
+    if new_source.url.len() == 0 {
+        return Err(ResourceError::InvalidInput("Url is empty".to_string()));
+    }
+
+    let url_regex = regex::Regex::new(r"^(https?|webcals)://(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)").unwrap();

```

Comment:
Can we put the inside of this, inside its own variable to make the lines smaller?

Reply:
Not really sure how we'd name it.

Something like this?
```rs
let regex_source = r"^(https?|webcals)://(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)";
let url_regex = regex::Regex::new(regex_source).unwrap();
```